src/globus_cli/commands/timer/_common.py

```diff
@@ -90,3 +107,81 @@ def isoformat_to_local(
     """,
     flags=re.VERBOSE,
 )
+
+
+def parse_timedelta(s: str) -> datetime.timedelta:
+    matches = timedelta_regex.match(s)
+    if not matches:
+        raise ValueError(f"couldn't parse timedelta from string: {s}")
+    groups = {k: int(v) for k, v in matches.groupdict(0).items()}
+    # timedelta accepts kwargs for units up through days, have to convert weeks
+    groups["days"] += groups.pop("weeks", 0) * 7
```

Comment:
This might be evidence of an older python version?
3.7 has `weeks=...` support:
https://github.com/python/cpython/blob/239b2d9b4703ea5301dab698fe922de997973476/Lib/datetime.py#L474

Overall though, I really like how this regex + parse function reads. It's very legible.

Reply:
Indeed, a relic of 3.6 support.